04.ls/ls.rb

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+COL_MAX = 3
+PADDING = 2
+
+def main
+  path = ARGV[0] if ARGV[0]
+  files = if path
+            collect_files(path) if FileTest.directory?(path)
+          else
+            collect_files
```

Comment:
```suggestion
            collect_files(path)
```

pathがnilでもそれをそのまま渡して collect_files 内でnilかどうかの判定をすれば良いので条件分岐しなくてもいいと思います

Reply:
> このメソッドは常に配列を返すようにして、このメソッドを呼ぶ前に正しいパスかどうかを判定するようなロジックの方が良いと思います。https://github.com/taco-nantai/ruby-practices/pull/4#discussion_r1429024776

先日頂いたこちらのレビューを見て、パスが正しいかどうかの判定は、メソッド(collect_files)を呼ぶ前に行う方がいいと解釈しました。
今回頂いたレビューの内容としては、パスが正しいかどうかの判定は、メソッド(collect_files)の中で行う方がいいという風に解釈したのですが、それで間違いないでしょうか。
仮にそうであった場合、パスが正しいかどうかの判定は、メソッドを呼ぶ前とメソッドの中、どちらで行えばよいでしょうか。
ご教授いただければ幸いです🙇‍♂️

Reply:
どのアプローチが絶対に正しい、というものはないのですが、個人的には collect_files は常に正しいパス（またはnil）が渡されるという前提にした方が良いと思います。

なので、collect_filesを呼ぶ前にpathが正しいかどうかを判定して、正しくなければその時点で処理を中断するのがいいと思います。

=====

あとは全然違う考え方なのですが、"正しいパスを入力してください"みたいな出力は必須要件ではないので、パスが正しいかどうかのチェックをしない（おかしなパスが渡されたらエラーが発生するが、そういうものとして許容する）というアプローチもあります。

実務ではおかしな入力にどう対処するか（いわゆる「異常系」の考慮）、というのは大事になるのですが、異常系に本気で対処しようとするとかなり考えることが増えてコード量も増えます（実際のところ、今考えることが増えてますよね？）。
コード量が増えると必要なテストパターンも増えます。

参考 https://bootcamp.fjord.jp/pages/write-code-write-test

lsコマンドのプラクティスはあくまで正常系のロジック力を鍛えるプラクティスであり、異常系の考慮はプラクティスのメインテーマではありません。

なので、僕としては「おかしなパスを指定されたらこのプログラムはエラーで落ちます」という形で提出されてもOKを出すつもりです。